src/components/Table/DataTable/compenents/el-data-table/utils/select-strategy.js

```diff
@@ -133,18 +150,20 @@ class StrategyPersistSelection extends StrategyAbstract {
     this.elDataTable.$emit('toggle-row-selection', isSelected, row)
     this.updateElTableSelection()
   }
+
   clearSelection() {
     this.elDataTable.selected = []
     this.updateElTableSelection()
   }
+
   /**
    * 将selected状态同步到el-table中
    */
   updateElTableSelection() {
     const { data, id, selected } = this.elDataTable
     const selectedIds = new Set(selected.map(r => r[id]))
 
-    this.elTable.clearSelection()
+    this.elTable?.clearSelection()
 
     data.forEach(row => {
       const shouldBeSelected = selectedIds.has(row[id])
```

Comment:
The current code does not show any irregularities but could be improved in terms of documentation and readability. Here are some suggested improvements:

1. Remove unnecessary curly braces and spaces between methods, make them more consistent.
2. Document method names properly with descriptions to improve understanding.

To optimize performance while keeping the same functionality:
- Use `map` instead of `forEach`, it's faster when dealing with arrays.
- If there needs to be an update event firing after each selection change operation (`onSelectionChange`), consider using hooks like `useEffect`.
- Consider refactoring selectors and state handling into a separate logic component to simplify implementation.